community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/IndexQuery.java

```diff
@@ -206,10 +200,16 @@ public IndexQueryType type()
         }
 
         @Override
-        public boolean test( Value value )
+        public boolean acceptsValue( Value value )
         {
             return value != null && value != Values.NO_VALUE;
         }
+
+        @Override
+        public boolean acceptsValueAt( PropertyCursor property )
+        {
+            return true;
```

Comment:
Because the fact that we have a `PropertyCursor` (pointing to a valid property record), means the the property exists, ergo the predicate is fulfilled.